File: frontend/src/types/report.ts
export interface ReportInfo {
  id: number;
  investigation_id: string;
  case_name: string;
  dump_filename: string | null;
  sha256_hash: string | null;
  filename: string;
  file_size: number;
  status: string;
  error_message: string | null;
  generated_at: string;
}

export interface ReportListResponse {
  items: ReportInfo[];
}

export interface ReportGenerateResponse extends ReportInfo {
  message: string;
}

export interface ReportDetailResponse extends ReportInfo {
  memory_dump_filename: string | null;
  investigation_status: string | null;
  total_plugins: number;
  successful_plugins: number;
  failed_plugins: number;
  total_evidence: number;
  investigation_duration: number;
}

File: backend/app/schemas/report.py
"""
Report API Schemas
"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class ReportInfo(BaseModel):
    """Metadata for a generated report."""

    id: int

    investigation_id: str

    case_name: str

    dump_filename: str | None = None

    sha256_hash: str | None = None

    filename: str

    file_size: int

    status: str

    error_message: str | None = None

    generated_at: datetime


class ReportListResponse(BaseModel):
    """List of generated reports."""

    items: list[ReportInfo]


class ReportGenerateResponse(ReportInfo):
    """Result of a report generation request."""

    message: str


class ReportDetailResponse(ReportInfo):
    """Full report details plus investigation statistics."""

    memory_dump_filename: str | None = None

    investigation_status: str | None = None

    total_plugins: int = 0

    successful_plugins: int = 0

    failed_plugins: int = 0

    total_evidence: int = 0

    investigation_duration: float = 0.0


# ==============================================================================
# Public Exports
# ==============================================================================

__all__ = [
    "ReportInfo",
    "ReportListResponse",
    "ReportGenerateResponse",
    "ReportDetailResponse",
]
